File: src/main/java/proto/games/matchmaker/support/Converters.java
package proto.games.matchmaker.support;

import proto.games.matchmaker.domain.player.Position;
import proto.games.matchmaker.domain.player.Rank;

public final class Converters {
    private Converters() {
    }

    public static Rank eloToRank(String elo) {
        switch (elo.trim()) {
            case "Ferro":
                return Rank.IRON;
            case "Bronze":
                return Rank.BRONZE;
            case "Prata":
                return Rank.SILVER;
            case "Ouro":
                return Rank.GOLD;
            case "Platina":
                return Rank.PLATINUM;
            case "Diamente":
                return Rank.DIAMOND;
            case "Mestre":
                return Rank.MASTER;
            case "Grão-Mestre":
                return Rank.GRANDMASTER;
            case "Desafiante":
                return Rank.CHALLENGER;
            default:
                return Rank.UNRANKED;
        }
    }

    public static Position rotaToPosition(String rota) {
        switch (rota.trim()) {
            case "Topo":
                return Position.TOP;
            case "Selva":
                return Position.JUNGLE;
            case "Meio":
                return Position.MIDDLE;
            case "Atirador":
                return Position.BOTTOM;
            case "Suporte":
            default:
                return Position.SUPPORT;

        }
    }
}

File: src/main/kotlin/proto/games/matchmaker/support/Converters.kt
package proto.games.matchmaker.support

import proto.games.matchmaker.domain.player.Position
import proto.games.matchmaker.domain.player.Rank

object Converters {
    fun eloToRank(elo: String): Rank {
        return when (elo.trim { it <= ' ' }) {
            "Ferro" -> Rank.IRON
            "Bronze" -> Rank.BRONZE
            "Prata" -> Rank.SILVER
            "Ouro" -> Rank.GOLD
            "Platina" -> Rank.PLATINUM
            "Diamente" -> Rank.DIAMOND
            "Mestre" -> Rank.MASTER
            "Grão-Mestre" -> Rank.GRANDMASTER
            "Desafiante" -> Rank.CHALLENGER
            else -> Rank.UNRANKED
        }
    }

    fun rotaToPosition(rota: String): Position {
        return when (rota.trim { it <= ' ' }) {
            "Topo" -> Position.TOP
            "Selva" -> Position.JUNGLE
            "Meio" -> Position.MIDDLE
            "Atirador" -> Position.BOTTOM
            "Suporte" -> Position.SUPPORT
            else -> Position.SUPPORT
        }
    }
}